toc/rfc/rfc-draft-bindable-secrets.md

```diff
@@ -122,19 +124,19 @@ So for a `json` type credential whose value is:
 }
 ```
 
-We would have a file called `application.yml` created under `/etc/my-conf`.
+The above `secret` would mount files named `application.yml` and `secret.txt` at the path configured in the binding.
 
-For a `value` type credential the file would simply take the same name as the `Secret Binding`.
+For a `value`-type `secret`, the file name be the `secret_binding`'s name.
 
-### Creating a `Secret`
+### Creating a `secret`
 
-To create a `Secret` we will introduce a new CLI command: `cf create-secret`.
+To create a `secret`, users will use a new CLI command: `cf create-secret`.
 
 ```console
 cf create-secret SECRET_NAME SECRET_TYPE -p <CREDENTIALS_INLINE_OR_FILE>
```

Comment:
I chose this for symmetry with other CF commands like `cf cups` or `cf bind-service`. It would distinguish between them the same way it does for those commands. If you want your secret value to be a path you could dump it in a file. 🙂 

```
cf cups -h
NAME:
   create-user-provided-service - Make a user-provided service instance available to CF apps

USAGE:
   cf create-user-provided-service SERVICE_INSTANCE [-p CREDENTIALS] [-l SYSLOG_DRAIN_URL] [-r ROUTE_SERVICE_URL] [-t TAGS]

   Pass comma separated credential parameter names to enable interactive mode:
   cf create-user-provided-service SERVICE_INSTANCE -p "comma, separated, parameter, names"

   Pass credential parameters as JSON to create a service non-interactively:
   cf create-user-provided-service SERVICE_INSTANCE -p '{"key1":"value1","key2":"value2"}'

   Specify a path to a file containing JSON:
   cf create-user-provided-service SERVICE_INSTANCE -p PATH_TO_FILE

EXAMPLES:
   cf create-user-provided-service my-db-mine -p "username, password"
   cf create-user-provided-service my-db-mine -p /path/to/credentials.json
   cf create-user-provided-service my-db-mine -t "list, of, tags"
   cf create-user-provided-service my-drain-service -l syslog://example.com
   cf create-user-provided-service my-route-service -r https://example.com
```

We could do separate flags, ala `ytt`, but is there precedent for that in the CF CLI?
```
 -v, --data-value stringArray                      Set specific data value to given value, as string (format: all.key1.subkey=123) (can be specified multiple times)
      --data-value-file stringArray                 Set specific data value to contents of a file (format: [@lib1:]all.key1.subkey={file path, HTTP URL, or '-' (i.e. stdin)}) (can be specified multiple times)
      --data-value-yaml stringArray                 Set specific data value to given value, parsed as YAML (format: all.key1.subkey=true) (can be specified multiple times)
      --data-values-env stringArray                 Extract data values (as strings) from prefixed env vars (format: PREFIX for PREFIX_all__key1=str) (can be specified multiple times)
      --data-values-env-yaml stringArray            Extract data values (parsed as YAML) from prefixed env vars (format: PREFIX for PREFIX_all__key1=true) (can be specified multiple times)
      --data-values-file stringArray                Set multiple data values via plain YAML files (format: [@lib1:]{file path, HTTP URL, or '-' (i.e. stdin)}) (can be specified multiple times)
```

Reply:
Okay, sold.